python/lsst/daf/butler/cli/cmd/commands.py

```diff
@@ -539,3 +541,55 @@ def collection_chain(**kwargs):
     """
     chain = script.collectionChain(**kwargs)
     print(f"[{', '.join(chain)}]")
+
+
+@click.command(cls=ButlerCommand)
+@repo_argument(required=True)
+@click.argument("dataset_type", required=True)
+@click.argument("run", required=True)
+@click.argument("table_file", required=True)
+@click.option("--formatter", type=str,
+              help="Fully-qualified python class to use as the Formatter. If not specified the formatter"
+              " will be determined from the dataset type and datastore configuration.")
+@click.option("--id-generation-mode",
+              default="UNIQUE",
+              help="Mode to use for generating dataset IDs. The default creates a unique ID. Other options"
+              " are: 'DATAID_TYPE' for creating a reproducible ID from the dataID;"
+              " 'DATAID_TYPE_RUN' for creating a reproducible ID from the dataID and run.  The latter is"
+              " usually used for 'raw'-type data.",
+              callback=to_upper,
+              type=click.Choice(("UNIQUE", "DATAID_TYPE", "DATAID_TYPE_RUN"), case_sensitive=False))
+@click.option("--data-id",
+              type=str,
+              multiple=True, callback=split_commas,
+              help="Keyword=value string with an additional dataId value that is fixed for all ingested"
+              " files. This can be used to simplify the table file by removing repeated entries that are"
+              " fixed for all files to be ingested.  Multiple key/values can be given either by using"
+              " comma separation or multiple command line options.")
+@click.option("--prefix",
+              type=str,
+              help="For relative paths in the table file, specify a prefix to use. The default is to"
+              " use the current working directory.")
+@transfer_option()
+def ingest_files(**kwargs):
+    """Ingest files from table file.
```

Comment:
Okay. I'll make it clear. For `butler ingest-raws` the dimension record for `exposure` can be created as part of ingest because the file metadata defines the dimension record. For general ingest we have no idea where to get the record information from -- for most use cases these ingests will be related to instrument/detector/exposure records so should already exist.